clients/python-wrapper/tests/integration/test_reference.py

```diff
@@ -47,6 +50,19 @@ def test_reference_merge_into(setup_branch_with_commits):
     repo = lakefs.Repository(branch.repo_id)
     main = repo.branch("main")
 
+    # test merging into same branch
```

Comment:
I find this test case disturbing 😆 
I wonder if there's a bug where we can merge a branch into itself

Reply:
Not sure it's a bug...
What error message would you expect?
And if so - would "allow empty" or "force" override it?

Reply:
I don't see any real use case for merging a ref into itself, at best it causes confusion for users and at worst can maybe lead to nastier bugs.
If you are unsure, you can always consult with @treeverse/product.
I don't think we should ever allow merging a branch into itself regardless of **any** flag. 
That's the baseline for me. In the future, if there will be any feature request with a compelling use-case we can start considering it and figure out all the implications